spec/integration/snapshot_query_examples_spec.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+# encoding: utf-8
+
+require 'spec_helper'
+
+describe 'Snapshot Query Examples' do
+  require_topology :replica_set, :sharded
+  require_no_auth
+  require_no_tls
+
+  min_server_fcv '5.0'
+
+  let(:uri_string) do
+    "mongodb://#{SpecConfig.instance.addresses.join(',')}/?w=majority"
+  end
+
+  context "Snapshot Query Example 1" do
+    before do
+      client = authorized_client.use('pets')
+      client['cats'].delete_many
+      client['dogs'].delete_many
+
+      client['cats'].insert_one(
+        name: "Whiskers",
+        color: "white",
+        age: 10,
+        adoptable: true
+      )
+
+      client['dogs'].insert_one(
+        name: "Pebbles",
+        color: "Brown",
+        age: 10,
+        adoptable: true
+      )
+      run_mongos_distincts "pets", "cats"
```

Comment:
`mongos_each_direct_client` does nothing when topology is not sharded. Can you try here

```
if ClusterConfig.instance.topology == :sharded
  run_mongos_distincts "pets", "cats"
else
  authorized_client.use('pets')['test'].distinct('foo')
  authorized_client.use('cats')['test'].distinct('foo')
end
```

Reply:
done!